hooks/useIsContract.ts

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+
+import { ChainFactory } from "../lib/chainFactory";
+import { EvmClientFactory } from "../lib/evmClient";
+
+const contractCache = new Map<string, boolean>();
+
+export function useIsContract(address: string) {
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
Whenever I write a statehook like this, it usually makes more sense to use `useQuery()`.

Reply:
Makes sense :+1:

Reply:
I took the opportunity to implement the contract check server side and make it cache for all users of our app instead.

Reply:
That's even better, cheers.